### PR TITLE
Add a JEP describing the jenkins/evergreen docker image

### PR DESCRIPTION
**Pull request by rtyler**
This has largely been prototyped and is thus far functional, but I think a
design document is still valuable for discussion and posterity.

### CODE REVIEW

File: jep/301/README.adoc [hunk 1]
```diff
@@ -0,0 +1,348 @@
...
+the Jenkins core process.
+
+
+[[evergreen-client]]
+=== Evergreen Client
+
+The second additional tool added into `jenkins/evergreen` is the
+`evergreen-client` process.  In short, it is the process responsible for
+managing updates, interrogating `supervisord` for process status, and reporting
+necessary telemetry to the Evergreen hosted service layer. The
```
Comment (orrc) [inline jep/301/README.adoc]: Should "Evergreen" here should be "Jenkins Essentials"?

File: jep/301/README.adoc [hunk 2]
```diff
@@ -0,0 +1,348 @@
...
+scripts committed upstream.
+
+
+[[plugins]]
+=== Plugins
+
+The `jenkins/evergreen` image does not have any plugins pre-installed, since
+the plugin versions in Jenkins Essentials will be constantly updating. Rather
+`jenkins/evergreen` must have some "first-boot" behavior to reach out to the
+Evergreen hosted service layer to request the current plugin versions.
```
Comment (orrc) [inline jep/301/README.adoc]: Same here.

File: jep/301/README.adoc [hunk 3]
```diff
@@ -0,0 +1,348 @@
...
+
+Since the `jenkins/evergreen` image is likely going to be moving slower than
+Jenkins Essentials itself, it will contain only the bare minimum amount of
+scripts to support configuring and setting up `evergreen-client`, Jenkins core,
+and `supervisord`.
+
+The scripts packaged into the image will not be responsible for configuring
+Jenkins or setting up the
+link:https://github.com/jenkinsci/jep/tree/master/jep/300#sane-defaults[Automatic Sane Defaults]
+descrbied in
```
Comment (orrc) [inline jep/301/README.adoc]: described

File: jep/301/README.adoc [hunk 4]
```diff
@@ -0,0 +1,348 @@
...
+
+== Reasoning
+
+As described in the <<motivation>> section, Jenkins Essentials requires a very
+confined and consistent environment. The requirements are a natural fit for
+Docker containers. Compared to three years ago, containers are now much more
+commonly accepted as a distribution mechanism for software such as Jenkins. As
+of this writing, the `jenkins/jenkins`
+footnote:[https://hub.docker.com/r/jenkins/jenkins/]
+image on Docker Hub has been "pulled" over 5 million times.
```
Comment (orrc) [inline jep/301/README.adoc]: Minor: should be "five" (as it's under 10, and consistent with "three" above)

File: jep/301/README.adoc (hunk 1 above)
Comment (rtyler) [reply jep/301/README.adoc]: No, I'm using the name "Evergreen" to describe the distribution mechanism in its entirety which covers client and server-side components

**Pull request merged**